engine/api/workflow_ascode.go

```diff
@@ -105,7 +105,7 @@ func (api *API) postWorkflowAsCodeHandler() service.Handler {
 		if err := workflow.RenameNode(ctx, api.mustDB(), &wf); err != nil {
 			return err
 		}
-		if err := workflow.IsValid(ctx, api.Cache, api.mustDB(), &wf, *p, workflow.LoadOptions{DeepPipeline: true}); err != nil {
+		if err := workflow.CompleteWorkflow(ctx, api.mustDB(), &wf, *p, workflow.LoadOptions{DeepPipeline: true}); err != nil {
```

Comment:
I think here we also need to check validity because the workflow comes from the request body.